src/Symfony/Component/Yaml/Parser.php

```diff
@@ -698,7 +707,7 @@ private function moveToPreviousLine(): bool
      */
     private function parseValue(string $value, int $flags, string $context)
     {
-        if (0 === strpos($value, '*')) {
+        if (0 === strncmp($value, '*', 1)) {
```

Comment:
`'' !== $value && '*' === $value[0]`?

Reply:
`strncmp` has the benefit that it only requires one condition, without the need to first verify that string offset `0` actually exists. Although I did not measure for this specifically it should be faster (at least when opcache is in effect).

Reply:
Comparing https://3v4l.org/UXmHA/vld#output and https://3v4l.org/p0nOT/vld#output suggests to me that using the function call requires one more operation if I understand that correctly.

Reply:
Although I don't know how the internals of `SEND_VAL` + `DO_ICALL` perform compared to an additional jump and `FETCH_DIM_R` it would seem that performing the comparisons 1,000,000 times shows `strncmp` consuming less memory and running faster except on PHP 7.4 which manages to eek out a tiny, tiny memory difference with advantage for the dual comparisons over `strncmp`.

It kind of makes sense in that `strncmp` does not have to extract the substring index and copy this memory before comparing.

Not sure how to proceed with this. Difference seems to be minimal. Future PHP versions may optimise opcode for the comparison case before optimising `strncmp` though...